core/services/relay/solana/chain_reader.go

```diff
@@ -0,0 +1,79 @@
+package solana
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	commonservices "github.com/smartcontractkit/chainlink-common/pkg/services"
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services"
+)
+
+const ServiceName = "SolanaChainReader"
+
+type SolanaChainReaderService struct {
+	lggr logger.Logger
+	commonservices.StateMachine
+}
+
+var (
+	ErrUnimplemented                         = errors.New("function unimplemented")
```

Comment:
There is a `types.UnimplementedError` type you should be able to import from chainlink-comon.

You can use it like:
`types.UnimplementedError("custom error message")`